salvo/src/lib/benchmark/binary_benchmark.py

```diff
@@ -0,0 +1,178 @@
+"""
+This module contains the methods to perform a Binary benchmark using
+containers for the scripts, nighthawk binaries, and envoy
+
+https://github.com/envoyproxy/nighthawk/blob/master/benchmarks/README.md
+"""
+
+import subprocess
+import logging
+import os
+
+import api.control_pb2 as proto_control
+import api.source_pb2 as proto_source
+
+from src.lib.benchmark import base_benchmark
+from src.lib.builder import (envoy_builder, nighthawk_builder)
+from src.lib import (cmd_exec, source_manager)
+
+log = logging.getLogger(__name__)
+
+
+class BinaryBenchmarkError(Exception):
+  """Error raised when running a binary benchmark in cases
+     where we cannot make progress due to abnormal conditions.
+  """
+
+class Benchmark(base_benchmark.BaseBenchmark):
+  """This benchmark class is the binary benchmark. We use a path to an Envoy
+     binary to execute the Nighthawk benchmarks using that specific build.
+  """
+
+  def __init__(
+    self, job_control: proto_control.JobControl, benchmark_name: str) -> None:
+    """ Initializes the benchmark class
+
+    Args:
+        job_control: The protobuf object containing the parameters and locations
+          of benchmark artifacts
+        benchmark_name: The name of the benchmark to execute
+
+    Raises:
```

Comment:
This Raises section may not be correct, since the initializer doesn't contain any `raise` statements. We only document raises that are directly invoked by the function we are describing.